beancount/query/query_compile.py

```diff
@@ -192,26 +192,41 @@ def __call__(self, context):
 class EvalMul(EvalBinaryOp):
 
     def __init__(self, left, right):
-        f = lambda x, y: Decimal(x * y)
-        super().__init__(f, left, right, Decimal)
+        if (left.dtype == amount.Amount) and (right.dtype in [int,float, Decimal]):
```

Comment:
I was thinking the output of 1/3 would be a float just like in python. But since we explicitly converting int/int to a decimal, then this isn't necessary.